PEP8 in test_expectmax

dipy/align/tests/test_expectmax.py:24:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:27:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:31:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:34:36: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:34:38: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:36:10: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:36:13: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:37:10: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:37:13: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:39:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:43:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:44:38: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:46:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:47:29: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:49:18: E222 multiple spaces after operator
dipy/align/tests/test_expectmax.py:51:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:54:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:55:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:56:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:59:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:64:29: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:64:37: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:64:38: E262 inline comment should start with '# '
dipy/align/tests/test_expectmax.py:64:80: E501 line too long (85 > 79 characters)
dipy/align/tests/test_expectmax.py:65:33: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:65:41: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:65:42: E262 inline comment should start with '# '
dipy/align/tests/test_expectmax.py:65:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:68:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:72:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:75:80: E501 line too long (105 > 79 characters)
dipy/align/tests/test_expectmax.py:76:80: E501 line too long (105 > 79 characters)
dipy/align/tests/test_expectmax.py:78:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:83:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:86:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:90:77: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:90:78: E262 inline comment should start with '# '
dipy/align/tests/test_expectmax.py:90:80: E501 line too long (118 > 79 characters)
dipy/align/tests/test_expectmax.py:91:32: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:93:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:102:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:104:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:104:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:104:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:108:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:110:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:110:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:110:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:114:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:116:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:116:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:116:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:120:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:122:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:122:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:122:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:126:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:128:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:128:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:128:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:148:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:151:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:155:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:159:36: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:159:38: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:161:10: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:161:13: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:162:10: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:162:13: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:163:10: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:163:13: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:165:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:169:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:170:38: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:172:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:173:29: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:175:18: E222 multiple spaces after operator
dipy/align/tests/test_expectmax.py:177:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:180:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:181:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:182:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:185:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:190:29: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:190:37: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:190:38: E262 inline comment should start with '# '
dipy/align/tests/test_expectmax.py:190:80: E501 line too long (85 > 79 characters)
dipy/align/tests/test_expectmax.py:191:33: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:191:41: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:191:42: E262 inline comment should start with '# '
dipy/align/tests/test_expectmax.py:191:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:194:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:198:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:201:80: E501 line too long (105 > 79 characters)
dipy/align/tests/test_expectmax.py:202:80: E501 line too long (105 > 79 characters)
dipy/align/tests/test_expectmax.py:203:80: E501 line too long (105 > 79 characters)
dipy/align/tests/test_expectmax.py:205:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:210:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:213:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:217:77: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:217:78: E262 inline comment should start with '# '
dipy/align/tests/test_expectmax.py:217:80: E501 line too long (118 > 79 characters)
dipy/align/tests/test_expectmax.py:218:32: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:218:56: E231 missing whitespace after ','
dipy/align/tests/test_expectmax.py:220:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:228:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:230:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:230:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:230:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:234:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:236:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:236:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:236:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:240:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:242:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:242:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:242:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:246:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:248:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:248:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:248:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:252:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:254:55: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:254:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:254:83: E225 missing whitespace around operator
dipy/align/tests/test_expectmax.py:258:1: E302 expected 2 blank lines, found 1
dipy/align/tests/test_expectmax.py:261:20: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:262:25: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:262:80: E501 line too long (88 > 79 characters)
dipy/align/tests/test_expectmax.py:268:48: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:268:50: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:270:80: E501 line too long (95 > 79 characters)
dipy/align/tests/test_expectmax.py:271:50: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:271:52: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:271:62: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:271:80: E501 line too long (100 > 79 characters)
dipy/align/tests/test_expectmax.py:272:80: E501 line too long (80 > 79 characters)
dipy/align/tests/test_expectmax.py:280:64: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:280:80: E501 line too long (129 > 79 characters)
dipy/align/tests/test_expectmax.py:281:80: E501 line too long (91 > 79 characters)
dipy/align/tests/test_expectmax.py:283:46: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:283:48: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:284:62: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:284:80: E501 line too long (93 > 79 characters)
dipy/align/tests/test_expectmax.py:285:44: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:286:55: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:286:80: E501 line too long (84 > 79 characters)
dipy/align/tests/test_expectmax.py:287:66: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:287:80: E501 line too long (95 > 79 characters)
dipy/align/tests/test_expectmax.py:297:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:301:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:304:80: E501 line too long (86 > 79 characters)
dipy/align/tests/test_expectmax.py:311:20: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:312:28: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:312:80: E501 line too long (91 > 79 characters)
dipy/align/tests/test_expectmax.py:318:48: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:318:50: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:320:80: E501 line too long (95 > 79 characters)
dipy/align/tests/test_expectmax.py:321:50: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:321:52: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:321:62: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:321:80: E501 line too long (100 > 79 characters)
dipy/align/tests/test_expectmax.py:322:80: E501 line too long (80 > 79 characters)
dipy/align/tests/test_expectmax.py:330:64: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:330:80: E501 line too long (129 > 79 characters)
dipy/align/tests/test_expectmax.py:331:80: E501 line too long (91 > 79 characters)
dipy/align/tests/test_expectmax.py:333:46: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:333:48: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:334:62: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:334:80: E501 line too long (93 > 79 characters)
dipy/align/tests/test_expectmax.py:335:44: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:336:55: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:336:80: E501 line too long (84 > 79 characters)
dipy/align/tests/test_expectmax.py:337:66: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:337:80: E501 line too long (95 > 79 characters)
dipy/align/tests/test_expectmax.py:347:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:351:80: E501 line too long (87 > 79 characters)
dipy/align/tests/test_expectmax.py:354:80: E501 line too long (86 > 79 characters)
dipy/align/tests/test_expectmax.py:363:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:366:21: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:368:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:373:80: E501 line too long (91 > 79 characters)
dipy/align/tests/test_expectmax.py:374:80: E501 line too long (88 > 79 characters)
dipy/align/tests/test_expectmax.py:376:32: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:376:34: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:378:61: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:378:63: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:379:59: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:379:61: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:381:1: E302 expected 2 blank lines, found 1
dipy/align/tests/test_expectmax.py:386:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:390:24: E261 at least two spaces before inline comment
dipy/align/tests/test_expectmax.py:392:5: E265 block comment should start with '# '
dipy/align/tests/test_expectmax.py:397:80: E501 line too long (94 > 79 characters)
dipy/align/tests/test_expectmax.py:398:80: E501 line too long (88 > 79 characters)
dipy/align/tests/test_expectmax.py:400:32: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:400:34: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:402:61: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:402:63: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:403:59: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:403:61: E251 unexpected spaces around keyword / parameter equals
dipy/align/tests/test_expectmax.py:406:12: E225 missing whitespace around operator

--- a/dipy/align/tests/test_expectmax.py
+++ b/dipy/align/tests/test_expectmax.py
@@ -75,10 +75,10 @@ def test_compute_em_demons_step_2d():
     # Pixels with gradient!=0 and sigma_i_sq=0
     sqnrm = sq_norm_grad_G[random_labels == 1]
     sigma_i_sq[random_labels == 1] = 0
-    expected[random_labels == 1, 0] = delta_field[
-        random_labels == 1] * grad_G[random_labels == 1, 0] / sqnrm
-    expected[random_labels == 1, 1] = delta_field[
-        random_labels == 1] * grad_G[random_labels == 1, 1] / sqnrm
+    expected[random_labels == 1, 0] = (delta_field[random_labels == 1] *
+                                       grad_G[random_labels == 1, 0] / sqnrm)
+    expected[random_labels == 1, 1] = (delta_field[random_labels == 1] *
+                                       grad_G[random_labels == 1, 1] / sqnrm)
 
     # Pixels with gradient=0 and sigma_i_sq=0
     sigma_i_sq[random_labels == 2] = 0
@@ -212,12 +212,12 @@ def test_compute_em_demons_step_3d():
     # Pixels with gradient!=0 and sigma_i_sq=0
     sqnrm = sq_norm_grad_G[random_labels == 1]
     sigma_i_sq[random_labels == 1] = 0
-    expected[random_labels == 1, 0] = delta_field[
-        random_labels == 1] * grad_G[random_labels == 1, 0] / sqnrm
-    expected[random_labels == 1, 1] = delta_field[
-        random_labels == 1] * grad_G[random_labels == 1, 1] / sqnrm
-    expected[random_labels == 1, 2] = delta_field[
-        random_labels == 1] * grad_G[random_labels == 1, 2] / sqnrm
+    expected[random_labels == 1, 0] = (delta_field[random_labels == 1] *
+                                       grad_G[random_labels == 1, 0] / sqnrm)
+    expected[random_labels == 1, 1] = (delta_field[random_labels == 1] *
+                                       grad_G[random_labels == 1, 1] / sqnrm)
+    expected[random_labels == 1, 2] = (delta_field[random_labels == 1] *
+                                       grad_G[random_labels == 1, 2] / sqnrm)
 
     # Pixels with gradient=0 and sigma_i_sq=0
     sigma_i_sq[random_labels == 2] = 0
@@ -308,8 +308,8 @@ def test_quantize_positive_2d():
 
     # make sure additive noise doesn't change the quantization result
     noise_amplitude = np.min([delta / 4.0, min_positive / 4.0])
-    noise = np.random.ranf(np.size(true_quantization)).reshape(
-        img_shape) * noise_amplitude
+    sz = np.size(true_quantization)
+    noise = np.random.ranf(sz).reshape(img_shape) * noise_amplitude
     noise = noise.astype(floating)
     input_image = np.ndarray(img_shape, dtype=floating)
     # assign intensities plus noise
@@ -372,8 +372,8 @@ def test_quantize_positive_3d():
 
     # make sure additive noise doesn't change the quantization result
     noise_amplitude = np.min([delta / 4.0, min_positive / 4.0])
-    noise = np.random.ranf(np.size(true_quantization)).reshape(
-        img_shape) * noise_amplitude
+    sz = np.size(true_quantization)
+    noise = np.random.ranf(sz).reshape(img_shape) * noise_amplitude
     noise = noise.astype(floating)
     input_image = np.ndarray(img_shape, dtype=floating)
     # assign intensities plus noise

--- a/dipy/align/tests/test_expectmax.py
+++ b/dipy/align/tests/test_expectmax.py
@@ -21,76 +21,82 @@ def test_compute_em_demons_step_2d():
                     image registration. NeuroImage, 45(1 Suppl), S61-72.
                     doi:10.1016/j.neuroimage.2008.10.040
     """
-    #Select arbitrary images' shape (same shape for both images)
+    # Select arbitrary images' shape (same shape for both images)
     sh = (30, 20)
 
-    #Select arbitrary centers
-    c_f = np.asarray(sh)/2
+    # Select arbitrary centers
+    c_f = np.asarray(sh) / 2
     c_g = c_f + 0.5
 
-    #Compute the identity vector field I(x) = x in R^2
+    # Compute the identity vector field I(x) = x in R^2
     x_0 = np.asarray(range(sh[0]))
     x_1 = np.asarray(range(sh[1]))
-    X = np.ndarray(sh + (2,), dtype = np.float64)
+    X = np.ndarray(sh + (2,), dtype=np.float64)
     O = np.ones(sh)
-    X[...,0]= x_0[:, None] * O
-    X[...,1]= x_1[None, :] * O
+    X[..., 0] = x_0[:, None] * O
+    X[..., 1] = x_1[None, :] * O
 
-    #Compute the gradient fields of F and G
+    # Compute the gradient fields of F and G
     grad_F = X - c_f
     grad_G = X - c_g
 
-    #The squared norm of grad_G to be used later
-    sq_norm_grad_G = np.sum(grad_G**2,-1)
+    # The squared norm of grad_G to be used later
+    sq_norm_grad_G = np.sum(grad_G**2, -1)
 
-    #Compute F and G
-    F = 0.5*np.sum(grad_F**2,-1)
-    G = 0.5*sq_norm_grad_G
-    delta_field =  G - F
+    # Compute F and G
+    F = 0.5 * np.sum(grad_F**2, -1)
+    G = 0.5 * sq_norm_grad_G
+    delta_field = G - F
 
-    #Now select an arbitrary parameter for $\sigma_x$ (eq 4 in [Vercauteren09])
+    # Now select an arbitrary parameter for
+    # $\sigma_x$ (eq 4 in [Vercauteren09])
     sigma_x_sq = 1.5
 
-    #Set arbitrary values for $\sigma_i$ (eq. 4 in [Vercauteren09])
-    #The original Demons algorithm used simply |F(x) - G(x)| as an
-    #estimator, so let's use it as well
+    # Set arbitrary values for $\sigma_i$ (eq. 4 in [Vercauteren09])
+    # The original Demons algorithm used simply |F(x) - G(x)| as an
+    # estimator, so let's use it as well
     sigma_i_sq = (F - G)**2
 
-    #Select some pixels to have special values
+    # Select some pixels to have special values
     np.random.seed(1346491)
-    random_labels = np.random.randint(0, 5, sh[0]*sh[1])
+    random_labels = np.random.randint(0, 5, sh[0] * sh[1])
     random_labels = random_labels.reshape(sh)
 
-    random_labels[sigma_i_sq==0] = 2 #this label is used to set sigma_i_sq == 0 below
-    random_labels[sq_norm_grad_G==0] = 2 #this label is used to set gradient == 0 below
+    # this label is used to set sigma_i_sq == 0 below
+    random_labels[sigma_i_sq == 0] = 2
+    # this label is used to set gradient == 0 below
+    random_labels[sq_norm_grad_G == 0] = 2
 
     expected = np.zeros_like(grad_G)
-    #Pixels with sigma_i_sq = inf
+    # Pixels with sigma_i_sq = inf
     sigma_i_sq[random_labels == 0] = np.inf
     expected[random_labels == 0, ...] = 0
 
-    #Pixels with gradient!=0 and sigma_i_sq=0
+    # Pixels with gradient!=0 and sigma_i_sq=0
     sqnrm = sq_norm_grad_G[random_labels == 1]
     sigma_i_sq[random_labels == 1] = 0
-    expected[random_labels == 1, 0] = delta_field[random_labels == 1]*grad_G[random_labels == 1, 0]/sqnrm
-    expected[random_labels == 1, 1] = delta_field[random_labels == 1]*grad_G[random_labels == 1, 1]/sqnrm
+    expected[random_labels == 1, 0] = delta_field[
+        random_labels == 1] * grad_G[random_labels == 1, 0] / sqnrm
+    expected[random_labels == 1, 1] = delta_field[
+        random_labels == 1] * grad_G[random_labels == 1, 1] / sqnrm
 
-    #Pixels with gradient=0 and sigma_i_sq=0
+    # Pixels with gradient=0 and sigma_i_sq=0
     sigma_i_sq[random_labels == 2] = 0
     grad_G[random_labels == 2, ...] = 0
     expected[random_labels == 2, ...] = 0
 
-    #Pixels with gradient=0 and sigma_i_sq!=0
+    # Pixels with gradient=0 and sigma_i_sq!=0
     grad_G[random_labels == 3, ...] = 0
 
-    #Directly compute the demons step according to eq. 4 in [Vercauteren09]
+    # Directly compute the demons step according to eq. 4 in [Vercauteren09]
     num = (sigma_x_sq * (F - G))[random_labels >= 3]
     den = (sigma_x_sq * sq_norm_grad_G + sigma_i_sq)[random_labels >= 3]
 
-    expected[random_labels >= 3] = -1 * np.array(grad_G[random_labels >= 3]) #This is $J^{P}$ in eq. 4 [Vercauteren09]
-    expected[random_labels >= 3,...] *= (num / den)[..., None]
+    # This is $J^{P}$ in eq. 4 [Vercauteren09]
+    expected[random_labels >= 3] = -1 * np.array(grad_G[random_labels >= 3])
+    expected[random_labels >= 3, ...] *= (num / den)[..., None]
 
-    #Now compute it using the implementation under test
+    # Now compute it using the implementation under test
 
     actual = np.empty_like(expected, dtype=floating)
     em.compute_em_demons_step_2d(np.array(delta_field, dtype=floating),
@@ -99,33 +105,38 @@ def test_compute_em_demons_step_2d():
                                  sigma_x_sq,
                                  actual)
 
-    #Test sigma_i_sq == inf
+    # Test sigma_i_sq == inf
     try:
-        assert_array_almost_equal(actual[random_labels==0], expected[random_labels==0])
+        assert_array_almost_equal(actual[random_labels == 0],
+                                  expected[random_labels == 0])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq == inf")
 
-    #Test sigma_i_sq == 0 and gradient != 0
+    # Test sigma_i_sq == 0 and gradient != 0
     try:
-        assert_array_almost_equal(actual[random_labels==1], expected[random_labels==1])
+        assert_array_almost_equal(actual[random_labels == 1],
+                                  expected[random_labels == 1])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq == 0 and gradient != 0")
 
-    #Test sigma_i_sq == 0 and gradient == 0
+    # Test sigma_i_sq == 0 and gradient == 0
     try:
-        assert_array_almost_equal(actual[random_labels==2], expected[random_labels==2])
+        assert_array_almost_equal(actual[random_labels == 2],
+                                  expected[random_labels == 2])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq == 0 and gradient == 0")
 
-    #Test sigma_i_sq != 0 and gradient == 0
+    # Test sigma_i_sq != 0 and gradient == 0
     try:
-        assert_array_almost_equal(actual[random_labels==3], expected[random_labels==3])
+        assert_array_almost_equal(actual[random_labels == 3],
+                                  expected[random_labels == 3])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq != 0 and gradient == 0 ")
 
-    #Test sigma_i_sq != 0 and gradient != 0
+    # Test sigma_i_sq != 0 and gradient != 0
     try:
-        assert_array_almost_equal(actual[random_labels==4], expected[random_labels==4])
+        assert_array_almost_equal(actual[random_labels == 4],
+                                  expected[random_labels == 4])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq != 0 and gradient != 0")
 
@@ -145,79 +156,86 @@ def test_compute_em_demons_step_3d():
                     doi:10.1016/j.neuroimage.2008.10.040
     """
 
-    #Select arbitrary images' shape (same shape for both images)
+    # Select arbitrary images' shape (same shape for both images)
     sh = (20, 15, 10)
 
-    #Select arbitrary centers
-    c_f = np.asarray(sh)/2
+    # Select arbitrary centers
+    c_f = np.asarray(sh) / 2
     c_g = c_f + 0.5
 
-    #Compute the identity vector field I(x) = x in R^2
+    # Compute the identity vector field I(x) = x in R^2
     x_0 = np.asarray(range(sh[0]))
     x_1 = np.asarray(range(sh[1]))
     x_2 = np.asarray(range(sh[2]))
-    X = np.ndarray(sh + (3,), dtype = np.float64)
+    X = np.ndarray(sh + (3,), dtype=np.float64)
     O = np.ones(sh)
-    X[...,0]= x_0[:, None, None] * O
-    X[...,1]= x_1[None, :, None] * O
-    X[...,2]= x_2[None, None, :] * O
+    X[..., 0] = x_0[:, None, None] * O
+    X[..., 1] = x_1[None, :, None] * O
+    X[..., 2] = x_2[None, None, :] * O
 
-    #Compute the gradient fields of F and G
+    # Compute the gradient fields of F and G
     grad_F = X - c_f
     grad_G = X - c_g
 
-    #The squared norm of grad_G to be used later
-    sq_norm_grad_G = np.sum(grad_G**2,-1)
+    # The squared norm of grad_G to be used later
+    sq_norm_grad_G = np.sum(grad_G**2, -1)
 
-    #Compute F and G
-    F = 0.5*np.sum(grad_F**2,-1)
-    G = 0.5*sq_norm_grad_G
-    delta_field =  G - F
+    # Compute F and G
+    F = 0.5 * np.sum(grad_F**2, -1)
+    G = 0.5 * sq_norm_grad_G
+    delta_field = G - F
 
-    #Now select an arbitrary parameter for $\sigma_x$ (eq 4 in [Vercauteren09])
+    # Now select an arbitrary parameter for
+    # $\sigma_x$ (eq 4 in [Vercauteren09])
     sigma_x_sq = 1.5
 
-    #Set arbitrary values for $\sigma_i$ (eq. 4 in [Vercauteren09])
-    #The original Demons algorithm used simply |F(x) - G(x)| as an
-    #estimator, so let's use it as well
+    # Set arbitrary values for $\sigma_i$ (eq. 4 in [Vercauteren09])
+    # The original Demons algorithm used simply |F(x) - G(x)| as an
+    # estimator, so let's use it as well
     sigma_i_sq = (F - G)**2
 
-    #Select some pixels to have special values
+    # Select some pixels to have special values
     np.random.seed(1346491)
-    random_labels = np.random.randint(0, 5, sh[0]*sh[1]*sh[2])
+    random_labels = np.random.randint(0, 5, sh[0] * sh[1] * sh[2])
     random_labels = random_labels.reshape(sh)
 
-    random_labels[sigma_i_sq==0] = 2 #this label is used to set sigma_i_sq == 0 below
-    random_labels[sq_norm_grad_G==0] = 2 #this label is used to set gradient == 0 below
+    # this label is used to set sigma_i_sq == 0 below
+    random_labels[sigma_i_sq == 0] = 2
+    # this label is used to set gradient == 0 below
+    random_labels[sq_norm_grad_G == 0] = 2
 
     expected = np.zeros_like(grad_G)
-    #Pixels with sigma_i_sq = inf
+    # Pixels with sigma_i_sq = inf
     sigma_i_sq[random_labels == 0] = np.inf
     expected[random_labels == 0, ...] = 0
 
-    #Pixels with gradient!=0 and sigma_i_sq=0
+    # Pixels with gradient!=0 and sigma_i_sq=0
     sqnrm = sq_norm_grad_G[random_labels == 1]
     sigma_i_sq[random_labels == 1] = 0
-    expected[random_labels == 1, 0] = delta_field[random_labels == 1]*grad_G[random_labels == 1, 0]/sqnrm
-    expected[random_labels == 1, 1] = delta_field[random_labels == 1]*grad_G[random_labels == 1, 1]/sqnrm
-    expected[random_labels == 1, 2] = delta_field[random_labels == 1]*grad_G[random_labels == 1, 2]/sqnrm
+    expected[random_labels == 1, 0] = delta_field[
+        random_labels == 1] * grad_G[random_labels == 1, 0] / sqnrm
+    expected[random_labels == 1, 1] = delta_field[
+        random_labels == 1] * grad_G[random_labels == 1, 1] / sqnrm
+    expected[random_labels == 1, 2] = delta_field[
+        random_labels == 1] * grad_G[random_labels == 1, 2] / sqnrm
 
-    #Pixels with gradient=0 and sigma_i_sq=0
+    # Pixels with gradient=0 and sigma_i_sq=0
     sigma_i_sq[random_labels == 2] = 0
     grad_G[random_labels == 2, ...] = 0
     expected[random_labels == 2, ...] = 0
 
-    #Pixels with gradient=0 and sigma_i_sq!=0
+    # Pixels with gradient=0 and sigma_i_sq!=0
     grad_G[random_labels == 3, ...] = 0
 
-    #Directly compute the demons step according to eq. 4 in [Vercauteren09]
+    # Directly compute the demons step according to eq. 4 in [Vercauteren09]
     num = (sigma_x_sq * (F - G))[random_labels >= 3]
     den = (sigma_x_sq * sq_norm_grad_G + sigma_i_sq)[random_labels >= 3]
 
-    expected[random_labels >= 3] = -1 * np.array(grad_G[random_labels >= 3]) #This is $J^{P}$ in eq. 4 [Vercauteren09]
-    expected[random_labels >= 3,...] *= (num / den)[...,None]
+    # This is $J^{P}$ in eq. 4 [Vercauteren09]
+    expected[random_labels >= 3] = -1 * np.array(grad_G[random_labels >= 3])
+    expected[random_labels >= 3, ...] *= (num / den)[..., None]
 
-    #Now compute it using the implementation under test
+    # Now compute it using the implementation under test
     actual = np.empty_like(expected, dtype=floating)
     em.compute_em_demons_step_3d(np.array(delta_field, dtype=floating),
                                  np.array(sigma_i_sq, dtype=floating),
@@ -225,66 +243,83 @@ def test_compute_em_demons_step_3d():
                                  sigma_x_sq,
                                  actual)
 
-    #Test sigma_i_sq == inf
+    # Test sigma_i_sq == inf
     try:
-        assert_array_almost_equal(actual[random_labels==0], expected[random_labels==0])
+        assert_array_almost_equal(actual[random_labels == 0],
+                                  expected[random_labels == 0])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq == inf")
 
-    #Test sigma_i_sq == 0 and gradient != 0
+    # Test sigma_i_sq == 0 and gradient != 0
     try:
-        assert_array_almost_equal(actual[random_labels==1], expected[random_labels==1])
+        assert_array_almost_equal(actual[random_labels == 1],
+                                  expected[random_labels == 1])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq == 0 and gradient != 0")
 
-    #Test sigma_i_sq == 0 and gradient == 0
+    # Test sigma_i_sq == 0 and gradient == 0
     try:
-        assert_array_almost_equal(actual[random_labels==2], expected[random_labels==2])
+        assert_array_almost_equal(actual[random_labels == 2],
+                                  expected[random_labels == 2])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq == 0 and gradient == 0")
 
-    #Test sigma_i_sq != 0 and gradient == 0
+    # Test sigma_i_sq != 0 and gradient == 0
     try:
-        assert_array_almost_equal(actual[random_labels==3], expected[random_labels==3])
+        assert_array_almost_equal(actual[random_labels == 3],
+                                  expected[random_labels == 3])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq != 0 and gradient == 0 ")
 
-    #Test sigma_i_sq != 0 and gradient != 0
+    # Test sigma_i_sq != 0 and gradient != 0
     try:
-        assert_array_almost_equal(actual[random_labels==4], expected[random_labels==4])
+        assert_array_almost_equal(actual[random_labels == 4],
+                                  expected[random_labels == 4])
     except AssertionError:
         raise AssertionError("Failed for sigma_i_sq != 0 and gradient != 0")
+
 
 def test_quantize_positive_2d():
     np.random.seed(1246592)
 
-    num_levels = 11 # an arbitrary number of quantization levels
-    img_shape = (15, 20) # arbitrary test image shape (must contain at least 3 elements)
+    # an arbitrary number of quantization levels
+    num_levels = 11
+    # arbitrary test image shape (must contain at least 3 elements)
+    img_shape = (15, 20)
     min_positive = 0.1
     max_positive = 1.0
     epsilon = 1e-8
 
     delta = (max_positive - min_positive + epsilon) / (num_levels - 1)
-    true_levels = np.zeros((num_levels,), dtype = np.float32)
+    true_levels = np.zeros((num_levels,), dtype=np.float32)
     # put the intensities at the centers of the bins
-    true_levels[1:] = np.linspace(min_positive+delta*0.5, max_positive-delta*0.5, num_levels-1)
-    true_quantization = np.empty(img_shape, dtype = np.int32) # generate a target quantization image
-    random_labels = np.random.randint(0, num_levels, np.size(true_quantization))
+    true_levels[1:] = np.linspace(min_positive + delta * 0.5,
+                                  max_positive - delta * 0.5, num_levels - 1)
+    # generate a target quantization image
+    true_quantization = np.empty(img_shape, dtype=np.int32)
+    random_labels = np.random.randint(0, num_levels,
+                                      np.size(true_quantization))
 
     # make sure there is at least one element equal to 0, 1 and num_levels-1
     random_labels[0] = 0
     random_labels[1] = 1
-    random_labels[2] = num_levels-1
+    random_labels[2] = num_levels - 1
     true_quantization[...] = random_labels.reshape(img_shape)
 
-    noise_amplitude = np.min([delta / 4.0, min_positive / 4.0]) # make sure additive noise doesn't change the quantization result
-    noise = np.random.ranf(np.size(true_quantization)).reshape(img_shape) * noise_amplitude
+    # make sure additive noise doesn't change the quantization result
+    noise_amplitude = np.min([delta / 4.0, min_positive / 4.0])
+    noise = np.random.ranf(np.size(true_quantization)).reshape(
+        img_shape) * noise_amplitude
     noise = noise.astype(floating)
-    input_image = np.ndarray(img_shape, dtype = floating)
-    input_image[...] = true_levels[true_quantization] + noise # assign intensities plus noise
-    input_image[true_quantization == 0] = 0 # preserve original zeros
-    input_image[true_quantization == 1] = min_positive # preserve min positive value
-    input_image[true_quantization == num_levels-1] = max_positive # preserve max positive value
+    input_image = np.ndarray(img_shape, dtype=floating)
+    # assign intensities plus noise
+    input_image[...] = true_levels[true_quantization] + noise
+    # preserve original zeros
+    input_image[true_quantization == 0] = 0
+    # preserve min positive value
+    input_image[true_quantization == 1] = min_positive
+    # preserve max positive value
+    input_image[true_quantization == num_levels - 1] = max_positive
 
     out, levels, hist = em.quantize_positive_2d(input_image, num_levels)
     levels = np.asarray(levels)
@@ -294,47 +329,61 @@ def test_quantize_positive_2d():
         current_bin = np.asarray(true_quantization == i).sum()
         assert_equal(hist[i], current_bin)
 
-    #test num_levels<2 and input image with zeros and non-zeros everywhere
+    # test num_levels<2 and input image with zeros and non-zeros everywhere
     assert_raises(ValueError, em.quantize_positive_2d, input_image, 0)
     assert_raises(ValueError, em.quantize_positive_2d, input_image, 1)
 
-    out, levels, hist = em.quantize_positive_2d(np.zeros(img_shape, dtype=floating), 2)
+    out, levels, hist = em.quantize_positive_2d(
+        np.zeros(img_shape, dtype=floating), 2)
     assert_equal(out, np.zeros(img_shape, dtype=np.int32))
 
-    out, levels, hist = em.quantize_positive_2d(np.ones(img_shape, dtype=floating), 2)
+    out, levels, hist = em.quantize_positive_2d(
+        np.ones(img_shape, dtype=floating), 2)
     assert_equal(out, np.ones(img_shape, dtype=np.int32))
 
 
 def test_quantize_positive_3d():
     np.random.seed(1246592)
 
-    num_levels = 11 # an arbitrary number of quantization levels
-    img_shape = (5, 10, 15) # arbitrary test image shape (must contain at least 3 elements)
+    # an arbitrary number of quantization levels
+    num_levels = 11
+    # arbitrary test image shape (must contain at least 3 elements)
+    img_shape = (5, 10, 15)
     min_positive = 0.1
     max_positive = 1.0
     epsilon = 1e-8
 
     delta = (max_positive - min_positive + epsilon) / (num_levels - 1)
-    true_levels = np.zeros((num_levels,), dtype = np.float32)
+    true_levels = np.zeros((num_levels,), dtype=np.float32)
     # put the intensities at the centers of the bins
-    true_levels[1:] = np.linspace(min_positive+delta*0.5, max_positive-delta*0.5, num_levels-1)
-    true_quantization = np.empty(img_shape, dtype = np.int32) # generate a target quantization image
-    random_labels = np.random.randint(0, num_levels, np.size(true_quantization))
+    true_levels[1:] = np.linspace(min_positive + delta * 0.5,
+                                  max_positive - delta * 0.5,
+                                  num_levels - 1)
+    # generate a target quantization image
+    true_quantization = np.empty(img_shape, dtype=np.int32)
+    random_labels = np.random.randint(0, num_levels,
+                                      np.size(true_quantization))
 
     # make sure there is at least one element equal to 0, 1 and num_levels-1
     random_labels[0] = 0
     random_labels[1] = 1
-    random_labels[2] = num_levels-1
+    random_labels[2] = num_levels - 1
     true_quantization[...] = random_labels.reshape(img_shape)
 
-    noise_amplitude = np.min([delta / 4.0, min_positive / 4.0]) # make sure additive noise doesn't change the quantization result
-    noise = np.random.ranf(np.size(true_quantization)).reshape(img_shape) * noise_amplitude
+    # make sure additive noise doesn't change the quantization result
+    noise_amplitude = np.min([delta / 4.0, min_positive / 4.0])
+    noise = np.random.ranf(np.size(true_quantization)).reshape(
+        img_shape) * noise_amplitude
     noise = noise.astype(floating)
-    input_image = np.ndarray(img_shape, dtype = floating)
-    input_image[...] = true_levels[true_quantization] + noise # assign intensities plus noise
-    input_image[true_quantization == 0] = 0 # preserve original zeros
-    input_image[true_quantization == 1] = min_positive # preserve min positive value
-    input_image[true_quantization == num_levels-1] = max_positive # preserve max positive value
+    input_image = np.ndarray(img_shape, dtype=floating)
+    # assign intensities plus noise
+    input_image[...] = true_levels[true_quantization] + noise
+    # preserve original zeros
+    input_image[true_quantization == 0] = 0
+    # preserve min positive value
+    input_image[true_quantization == 1] = min_positive
+    # preserve max positive value
+    input_image[true_quantization == num_levels - 1] = max_positive
 
     out, levels, hist = em.quantize_positive_3d(input_image, num_levels)
     levels = np.asarray(levels)
@@ -344,14 +393,16 @@ def test_quantize_positive_3d():
         current_bin = np.asarray(true_quantization == i).sum()
         assert_equal(hist[i], current_bin)
 
-    #test num_levels<2 and input image with zeros and non-zeros everywhere
+    # test num_levels<2 and input image with zeros and non-zeros everywhere
     assert_raises(ValueError, em.quantize_positive_3d, input_image, 0)
     assert_raises(ValueError, em.quantize_positive_3d, input_image, 1)
 
-    out, levels, hist = em.quantize_positive_3d(np.zeros(img_shape, dtype=floating), 2)
+    out, levels, hist = em.quantize_positive_3d(np.zeros(img_shape,
+                                                         dtype=floating), 2)
     assert_equal(out, np.zeros(img_shape, dtype=np.int32))
 
-    out, levels, hist = em.quantize_positive_3d(np.ones(img_shape, dtype=floating), 2)
+    out, levels, hist = em.quantize_positive_3d(np.ones(img_shape,
+                                                        dtype=floating), 2)
     assert_equal(out, np.ones(img_shape, dtype=np.int32))
 
 
@@ -360,54 +411,60 @@ def test_compute_masked_class_stats_2d():
 
     shape = (32, 32)
 
-    #Create random labels
+    # Create random labels
     labels = np.ndarray(shape, dtype=np.int32)
     labels[...] = np.random.randint(2, 10, np.size(labels)).reshape(shape)
-    labels[0, 0] = 1 # now label 0 is not present and label 1 occurs once
+    # now label 0 is not present and label 1 occurs once
+    labels[0, 0] = 1
 
-    #Create random values
+    # Create random values
     values = np.random.randn(shape[0], shape[1]).astype(floating)
     values *= labels
     values += labels
 
-    expected_means = [0, values[0, 0]] + [values[labels == i].mean() for i in range(2, 10)]
-    expected_vars = [np.inf, np.inf] + [values[labels == i].var() for i in range(2, 10)]
+    expected_means = [0, values[0, 0]] + \
+        [values[labels == i].mean() for i in range(2, 10)]
+    expected_vars = [np.inf, np.inf] + \
+        [values[labels == i].var() for i in range(2, 10)]
 
-    mask = np.ones(shape, dtype = np.int32)
+    mask = np.ones(shape, dtype=np.int32)
     means, vars = em.compute_masked_class_stats_2d(mask, values, 10, labels)
-    assert_array_almost_equal(means, expected_means, decimal = 4)
-    assert_array_almost_equal(vars, expected_vars, decimal = 4)
+    assert_array_almost_equal(means, expected_means, decimal=4)
+    assert_array_almost_equal(vars, expected_vars, decimal=4)
+
 
 def test_compute_masked_class_stats_3d():
     np.random.seed(1246592)
 
     shape = (32, 32, 32)
 
-    #Create random labels
+    # Create random labels
     labels = np.ndarray(shape, dtype=np.int32)
     labels[...] = np.random.randint(2, 10, np.size(labels)).reshape(shape)
 
-    labels[0, 0, 0] = 1 # now label 0 is not present and label 1 occurs once
+    # now label 0 is not present and label 1 occurs once
+    labels[0, 0, 0] = 1
 
-    #Create random values
+    # Create random values
     values = np.random.randn(shape[0], shape[1], shape[2]).astype(floating)
     values *= labels
     values += labels
 
-    expected_means = [0, values[0, 0, 0]] + [values[labels == i].mean() for i in range(2, 10)]
-    expected_vars = [np.inf, np.inf] + [values[labels == i].var() for i in range(2, 10)]
+    expected_means = [0, values[0, 0, 0]] + \
+        [values[labels == i].mean() for i in range(2, 10)]
+    expected_vars = [np.inf, np.inf] + \
+        [values[labels == i].var() for i in range(2, 10)]
 
-    mask = np.ones(shape, dtype = np.int32)
+    mask = np.ones(shape, dtype=np.int32)
     means, vars = em.compute_masked_class_stats_3d(mask, values, 10, labels)
-    assert_array_almost_equal(means, expected_means, decimal = 4)
-    assert_array_almost_equal(vars, expected_vars, decimal = 4)
+    assert_array_almost_equal(means, expected_means, decimal=4)
+    assert_array_almost_equal(vars, expected_vars, decimal=4)
 
 
-if __name__=='__main__':
+if __name__ == '__main__':
     test_compute_em_demons_step_2d()
     test_compute_em_demons_step_3d()
     test_quantize_positive_2d()
     test_quantize_positive_3d()
     test_compute_masked_class_stats_2d()
     test_compute_masked_class_stats_3d()
-

